src/VirtualClient/VirtualClient.Actions/Sysbench/SysbenchClientExecutor.cs

```diff
@@ -242,5 +261,78 @@ private async Task RunTPCCWorkloadAsync(EventContext telemetryContext, Cancellat
+        {
+            string arguments = $"{this.packageDirectory}/truncate-tables.py --dbName {this.DatabaseName}";
+
+            if (this.IsMultiRoleLayout())
+            {
+                ClientInstance instance = this.Layout.GetClientInstance(this.AgentId);
+                string serverIps = (instance.Role == ClientRole.Server) ? "localhost" : this.GetServerIpAddress();
+                arguments += $" --clientIps \"{serverIps}\"";
+            }
+
+            using (IProcessProxy process = await this.ExecuteCommandAsync(
+                PythonCommand,
+                arguments,
+                Environment.CurrentDirectory,
+                telemetryContext,
+                cancellationToken))
+            {
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    await this.LogProcessDetailsAsync(process, telemetryContext, "Sysbench", logToFile: true);
+                    process.ThrowIfDependencyInstallationFailed(process.StandardError.ToString());
+                }
+            }
+        }
+
+        private async Task PrepareOLTPMySQLDatabase(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            int tableCount = GetTableCount(this.DatabaseScenario, this.TableCount, this.Workload);
+            int threadCount = GetThreadCount(this.SystemManager, this.DatabaseScenario, this.Threads);
+            int recordCount = GetRecordCount(this.SystemManager, this.DatabaseScenario, this.RecordCount);
+
+            this.sysbenchPrepareArguments = $"--dbName {this.DatabaseName} --databaseSystem {this.DatabaseSystem} --benchmark {this.Benchmark} --tableCount {tableCount} --recordCount {recordCount} --threadCount {threadCount} --password {this.SuperUserPassword}";
+
+            if (this.IsMultiRoleLayout())
+            {
+                ClientInstance instance = this.Layout.GetClientInstance(this.AgentId);
+                string serverIp = (instance.Role == ClientRole.Server) ? "localhost" : this.GetServerIpAddress();
+                this.sysbenchPrepareArguments += $" --host \"{serverIp}\"";
+            }
+
+            string command = $"python3";
+            string arguments = $"{this.SysbenchPackagePath}/populate-database.py ";
+
+            using (IProcessProxy process = await this.ExecuteCommandAsync(
+                command,
+                arguments + this.sysbenchPrepareArguments,
+                this.SysbenchPackagePath,
+                telemetryContext,
+                cancellationToken))
+            {
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    await this.LogProcessDetailsAsync(process, telemetryContext, "Sysbench", logToFile: true);
+                    process.ThrowIfErrored<WorkloadException>(process.StandardError.ToString(), ErrorReason.WorkloadUnexpectedAnomaly);
+                }
+            }
+        }
+
+        private string GetServerIpAddress()
```

Comment:
I see 4 instance of this method in code base. Could you please make this into an extension method on EnvironmentLayout class and apply to other classes as well?